### PR TITLE
feat: migrate to manifest.toml

### PR DESCRIPTION
**Pull request by zmitchell**
## Proposed Changes

This PR officially migrates from `flox.nix` to `manifest.toml` for the manifest format. This entails creating a new environment template stored in the new `assets` directory (a new directory is used since we're also deleting `flox-bash`, which contains the previous template). `flox init` will now create a `manifest.toml` file under `.flox/env` and will no longer create a `flox.nix` file.

This change breaks 11 tests related to `flox list`, `flox install`, and `flox edit`:
- The `flox edit` tests require programmatic modification of the manifest. This was previously done using `nix-editor`, but better tooling exists for TOML files so we can use existing tools like [toml-cli](https://github.com/gnprice/toml-cli) to write a non-interactive editor if we so choose. Fixing these tests is left to a separate ticket.
- The `flox install` and `flox uninstall` functionality will be fixed in subsequent PRs.
- The `flox list` tests rely on both `flox install` and interpreting the manifest. These will be fixed as part of the `flox install` PR.
- The last failing test uses several deprecated commands which will be removed shortly, so root-causing that failure was considered a waste of time.

**Notes for review: This PR is based on flox#328 and will be rebased once that lands. Both this PR and #328 are being held back until the release is complete, but this PR is otherwise ready for review.**

## Current Behavior

`flox init` creates a `flox.nix` manifest and several more commands work.

## Checks

- [x] All tests pass.
- [ ] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [ ] I have accepted the flox [Contributor License Agreement](../blob/main/.github/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](../blob/main/.github/CONTRIBUTORS.csv) file by way of this pull request or one done previously.

## Release Notes

`flox edit`, `flox list`, `flox install`, and `flox uninstall` will all break. `flox init` will now create a `manifest.toml` instead of `flox.nix`.

### CODE REVIEW

File: assets/templateFloxEnv/flake.nix
Comment (ysndr) [inline assets/templateFloxEnv/flake.nix:1]: we shouldn't need anything but the manifest.toml
the flake.nix is being replaced by a nix lib shipped with flox AFAIK

Comment (zmitchell) [reply assets/templateFloxEnv/flake.nix:1]: Does that lib exist yet?

**Review by aakropotkin — APPROVED** [review]

File: assets/templateFloxEnv/flake.nix
Comment (mkenigs) [reply assets/templateFloxEnv/flake.nix:1]: No

Comment (zmitchell) [reply assets/templateFloxEnv/flake.nix:1]: So then let's worry about that later

Comment (mkenigs) [reply assets/templateFloxEnv/flake.nix:1]: +1

**Pull request merged**